### DISCUSSION TITLE
how to use ethers.js with sign api html.

### DISCUSSION THREAD

**Discussion by degenSumi** [WalletConnect/walletconnect-monorepo#2722] [Developer Support]
const connection = await web3Modal.connect({
        requiredNamespaces: {
          eip155: {
            methods: ['eth_sendTransaction'],
            chains: ['eip155:1','eip155:5' ],
            events: ['chainChanged', 'accountsChanged']
          }
        }
      });
      console.log(typeof(connection),connection);
      setWebmodal(connection);
      const provider = new ethers.providers.Web3Provider(connection);
I am trying to use ethers with web3modal/sign-html but it says error
Argument of type 'Struct' is not assignable to parameter of type 'ExternalProvider | JsonRpcFetchFunc'.t

**Comment by glitch-txs** [comment]
You can use ethers with the ethereum-provider library instead 
https://docs.walletconnect.com/2.0/web/providers/ethereum
